iamctl/pkg/utils/keywordReplacement.go

```diff
@@ -0,0 +1,237 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+func ReplaceKeywords(fileContent string, keywordMapping map[string]interface{}) string {
+
+	// Loop over the keyword mapping and replace each keyword in the file.
+	for keyword, value := range keywordMapping {
+		if value, ok := value.(string); ok {
+			fileContent = strings.ReplaceAll(fileContent, "{{"+keyword+"}}", value)
+		} else {
+			log.Printf("Keyword value for %s is not a string", keyword)
+		}
+	}
+	return fileContent
+}
+
+func AddKeywords(exportedData []byte, localFileData []byte, keywordMapping map[string]interface{}) []byte {
+
+	var localYaml interface{}
+	err := yaml.Unmarshal(localFileData, &localYaml)
+	if err != nil {
+		log.Println("Error: ", err)
+	}
+	// If the local file is empty or not available, return the exported data as it is.
+	if err == nil && localYaml != nil {
+		// Get keyword locations in local file.
+		keywordLocations := GetKeywordLocations(localYaml, []string{}, keywordMapping)
+		// Load exported app data as a yaml object.
+		var exportedYaml interface{}
+		err = yaml.Unmarshal(exportedData, &exportedYaml)
+		if err != nil {
+			log.Println("Error: ", err)
+		}
+
+		// Compare the fields with keywords in the exported file and the local file and modify the exported file.
+		exportedYaml = ModifyFieldsWithKeywords(exportedYaml, localYaml, keywordLocations, keywordMapping)
+
+		exportedData, err = yaml.Marshal(exportedYaml)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return exportedData
+}
+
+func GetKeywordLocations(fileData interface{}, path []string, keywordMapping map[string]interface{}) []string {
+
+	var keys []string
+	switch v := fileData.(type) {
+	case map[interface{}]interface{}:
+		for k, val := range v {
+			newPath := append(path, fmt.Sprintf("%v", k))
+			keys = append(keys, GetKeywordLocations(val, newPath, keywordMapping)...)
+		}
+	case []interface{}:
+		for _, val := range v {
+			arrayElementPath, err := resolvePathWithIdentifiers(path[len(path)-1], val, appArrayIdentifiers)
+			if err != nil {
+				log.Println("Error: ", err)
+				break
+			} else {
+				newPath := append(path, arrayElementPath)
+				keys = append(keys, GetKeywordLocations(val, newPath, keywordMapping)...)
+			}
+		}
+	case string:
+		if ContainsKeywords(fileData.(string), keywordMapping) {
+			thisPath := strings.Join(path, ".")
+			keys = append(keys, thisPath)
+		}
+	}
+	return keys
+}
+
+func resolvePathWithIdentifiers(arrayName string, element interface{}, identifiers map[string]string) (string, error) {
```

Comment:
can't we use this multi value return in other functions as well `(string, error)`?
This looks cleaner and lets the control flow handle respond to the error rather than doing that in the function itself